### PR TITLE
fix: implement URL encoding for invalid URI characters and add utility tests

### PR DESCRIPTION
**Pull request by grigoriev**
### Proposed changes

- Encodes invalid characters in relative URLs during HTML processing.
- Introduced `UrlUtils` class for URI encoding logic.
- Added comprehensive `UrlUtilsTest` to cover edge cases.

### Checklist

Before creating a PR, run through this checklist and mark each as complete:
- [x] I have read the [`CONTRIBUTING`](CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] I have updated any relevant documentation

### CODE REVIEW

File: src/main/java/ch/sbb/polarion/extension/docx_exporter/util/UrlUtils.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+public class UrlUtils {
+
+    /**
+     * Encodes characters that are invalid in URIs (like spaces) while preserving already encoded sequences
+     * and valid URI structure characters (like /, ?, #, =, &amp;).
+     *
+     * @param url the URL string to encode
+     * @return the URL with invalid characters encoded, or null/empty if input was null/empty
+     */
+    public static @Nullable String encodeInvalidUriCharacters(@Nullable String url) {
```
[the remark this answers is not in the document]
Comment (grigoriev) [reply src/main/java/ch/sbb/polarion/extension/docx_exporter/util/UrlUtils.java]: Renamed

File: src/test/java/ch/sbb/polarion/extension/docx_exporter/util/HtmlProcessorTest.java [hunk 2]
```diff
@@ -582,21 +567,24 @@ void testIsRelativeLink() {
...
-    @Test
-    void testResolveUrl() {
-        String baseUrl = "http://example.com/base/";
-        String rel = "foo/bar.html";
-        String result = processor.resolveUrl(baseUrl, rel);
-        assertEquals("http://example.com/base/foo/bar.html", result);
+    @ParameterizedTest
+    @CsvSource({
+            "http://example.com/base/, foo/bar.html, http://example.com/base/foo/bar.html",
+            "http://example.com, /polarion/#/project/MyProject/wiki/Common/My Document Title, http://example.com/polarion/#/project/MyProject/wiki/Common/My%20Document%20Title",
```
[the remark this answers is not in the document]
Comment (grigoriev) [reply src/test/java/ch/sbb/polarion/extension/docx_exporter/util/HtmlProcessorTest.java]: Tests extended

**Review by yurtsevich-sbb — APPROVED** [review]
